automatic_push/utils.py

```diff
@@ -16,79 +16,249 @@
+            # Probably revision filename contained invalid characters such as alphanumeric characters
+            logger.exception(f'Failed to process revision for {nested_filename}: (rev: {revision})')
+            revisions.append(revision)
+        if len(revisions) == 0:  # Single revision...
+            continue
+        # Keep the latest (max) revision and delete the rest
+        latest = revisions.index(max(revisions))
+        files_to_delete.remove(files_to_delete[latest])
+        for file_to_delete in files_to_delete:
+            if 'iana-if-type' in file_to_delete:
+                break
+            if os.path.exists((path := os.path.join(directory, file_to_delete))):
+                os.remove(path)
+
+
+def _get_module_name(filename: str) -> str:
+    return filename.split('.yang')[0].split('@')[0]
+
+
+def _is_revision_part_valid(revision_part: str) -> bool:
+    return revision_part.startswith('.') or revision_part.startswith('@')
+
+
+def extract_rfc_tgz(tgz_path: str, extract_to: str, logger: logging.Logger) -> bool:
+    """
+    Extract downloaded rfc.tgz file to directory and remove file.
+
+    Arguments:
+        :param tgz_path     (str) full path to the rfc.tgz file
+        :param extract_to   (str) path to the directory where rfc.tgz is extractracted to
+        :param logger       (logging.Logger) formated logger with the specified name
+    :return  (bool) Indicates if the tar archive was opened or not
+    """
+    try:
+        tgz = tarfile.open(tgz_path)
+        tar_opened = True
+        tgz.extractall(extract_to)
+        tgz.close()
+    except tarfile.ReadError:
+        tar_opened = False
+        logger.warning(
+            'Tarfile could not be opened. It might not have been generated yet. '
+            'Did the module-compilation cron job run already?',
+        )
+    os.remove(tgz_path)
+
+    return tar_opened
+
+
+@dataclass
+class PushResult:
+    is_successful: bool
+    detail: str
+
+
+def push_untracked_files(
+    repo: Repo,
+    commit_message: str,
+    logger: logging.Logger,
+    verified_commits_file_path: str,
+    is_production: bool,
+) -> PushResult:
+    """
+    Commits locally and pushes all the changes to the remote repository.
+
+    Arguments:
+          :param repo (Repo) Repository where to push changes.
+          :param commit_message (str) New commit message.
+          :param logger (logging.Logger) Logger instance to log information/exceptions.
+          :param verified_commits_file_path (str) Path to file where our verified commits should be stored
+          in order to verify commits in GitHub webhooks.
+          :param is_production (bool) If set to False then files would only be committed and not pushed to the repo.
+    :return (PushResult) Returns information about the push.
+    """
+    try:
+        untracked_files = repo.untracked_files
+        logger.info('Committing all files locally')
+        repo.git.commit(a=True, m=commit_message)
```

Comment:
[The -a option only stages changes to files that are already tracked in the repository. Any new files that you've added to the repository will still need to be explicitly staged using git add before they can be committed](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--a), so there should be a ```git add .``` command first